src/java.desktop/macosx/classes/sun/lwawt/macosx/CPrinterJob.java

```diff
@@ -251,6 +253,21 @@ private void setPrintToFile(boolean printToFile) {
         isPrintToFile = printToFile;
     }
 
+    private void setDestinationFile(String dest) {
+        if (attributes != null && dest != null) {
+            try {
+               URI destURI = new URI(dest);
+               attributes.add(new Destination(destURI));
+               destinationAttr = "" + destURI.getSchemeSpecificPart();
```

Comment:
This destinationAttr in the RasterPrinterJob is usually assigned to:
                destinationAttr = "" + new File(destination.getURI().
                                                getSchemeSpecificPart());
Do we need to do the same here?

Reply:
I see that in WPrinterJob.java we do the same as I am doing here.
I think the difference in RPJ may bethat we want to get a full path name which is what the new File did.
Here macOS gives us a full path name already. So I don't think it is needed.
It gave the required value for sure.